File: bubblesort.java
// Bubble Short Algorithm in java
public class BubbleSortExample 
{
    public static void main(String[] args) 
    {
    
        Integer[] array = new Integer[] { 12, 13, 24, 10, 3, 6, 90, 70 };
 
      
        bubbleSort(array, 0, array.length);
 
     
        System.out.println(Arrays.toString(array));
    }
 
    @SuppressWarnings({ "rawtypes", "unchecked" })
    public static void bubbleSort(Object[] array, int fromIndex, int toIndex) 
    {
        Object d;
        for (int i = toIndex - 1; i > fromIndex; i--) 
        {
            boolean isSorted = true;
            for (int j = fromIndex; j < i; j++) 
            {
                //If elements in wrong order then swap them
                if (((Comparable) array[j]).compareTo(array[j + 1]) > 0) 
                {
                    isSorted = false;
                    d = array[j + 1];
                    array[j + 1] = array[j];
                    array[j] = d;
                }
            }
            //If no swapping then array is already sorted
            if (isSorted)
                break;
        }
    }
}
 
// Output: [3, 6, 10, 12, 13, 24, 70, 90]

File: bubblesort.cpp
#include <iostream>  // std::cout
#include <algorithm> // std::swap

void bubbleSort(int arr[], int n)
{
    int i, j;
    for (i = 0; i < n - 1; ++i)
        for (j = 0; j < n - i - 1; ++j)
            if (arr[j] > arr[j + 1])
                std::swap(arr[j], arr[j + 1]);
}
int main()
{
    int arr[] = {5, 7, 2, 4, 9, 3, 1, 8, 6, 0};
    int n = sizeof(arr) / sizeof(arr[0]);
    bubbleSort(arr, n);
    std::cout << "Sorted array: ";
    for (int i = 0; i < n; ++i)
        std::cout << arr[i] << ' ';
    return 0;
}
